### PR TITLE
added Default folder

### PR DESCRIPTION
**Pull request by gabyx**

### CODE REVIEW

**Comment by gabyx** [comment]
DEFAULT_PARENT_DIR makes it usefull to specify add a location where the Configuration Files go, and defaulted to ${CMAKE_BINARY_DIR}/thirdparty  

all stuff goes into there by default which is cleaner

File: DownloadProject.cmake [hunk 1]
```diff
@@ -94,29 +95,35 @@ function(download_project)
         unset(OUTPUT_QUIET)
         message(STATUS "Downloading/updating ${DL_ARGS_PROJ}")
     endif()
-
+    
+    if(NOT DEFAULT_PARENT_DIR)
+       set(DEFAULT_PARENT_DIR  "${CMAKE_BINARY_DIR}/thirdparty" )
+    endif()
```
Comment (audiofanatic) [inline DownloadProject.cmake]: In this test, DEFAULT_PARENT_DIR will always be empty (unless you were unlucky and something at parent scope set it). You probably meant to use DL_ARGS_DEFAULT_PARENT_DIR, since that's what will be populated by `cmake_parse_arguments()`.

File: DownloadProject.CMakeLists.cmake.in [hunk 2]
```diff
@@ -2,6 +2,7 @@ cmake_minimum_required(VERSION 2.8.2)
 
 project(${DL_ARGS_PROJ}-download NONE)
 
+
```
Comment (audiofanatic) [inline DownloadProject.CMakeLists.cmake.in]: This is an unnecessary change.

**Comment by audiofanatic** [comment]
Thanks for the suggested changes. I understand the motivation behind this pull request, but there are some wider aspects to consider too:

- There are already people using this project, so changing the default download location may break their builds and make you a bit unpopular. ;) I'd prefer to keep the default behaviour the same unless there was a good reason to change it.
- While one main use case revolves around the external project being a third party project, there's also a common use case for using this for internal projects too. For example, it can be used to bring in other git repositories without having to use git submodules.
- You can already control the source and binary directories with the SOURCE_DIR and BINARY_DIR options. You can probably achieve the flexibility you want by explicitly handling a DOWNLOAD_DIR option as well (the name is set by ExternalProject just like SOURCE_DIR and BINARY_DIR) which controls just the directory into which the source is downloaded. Have a look at the [ExternalProject documentation](https://cmake.org/cmake/help/latest/module/ExternalProject.html) to see what these options all map into.

Would you consider modifying your patch to use DOWNLOAD_DIR instead and then just treat this in the same way as SOURCE_DIR and BINARY_DIR? Calling code would still need to set all three, but I think this helps preserve the more direct mapping to ExternalProject, which will make future maintainability easier and more robust.

**Comment by audiofanatic** [comment]
Actually, I can see it may be a bit more complicated than just adding the DOWNLOAD_DIR option, but explore the general intent and see what you come up with.

**Comment by audiofanatic** [comment]
As part of fixing the issue identified in #4, I've incorporated a modified version of your code which is a little more consistent with existing ExternalProject keywords. Hopefully it also meets your needs. I'll close this pull request, but if you encounter further related issues, open a bug report or comment on #4. Thanks.
